tests/nightly_tests/deploy.sh

```diff
@@ -79,42 +104,76 @@ echo "Deploying Cloudformation stack..." >> nightly_output.txt
-    echo "Marketplace Item                | Version"
-    echo "--------------------------------+--------"
-    echo "$config_content" | grep -E '^\s*-' | sed -E 's/^\s*-\s*name:\s*(.*)/\1/' | while read -r line; do
-        name=$(echo "$line" | cut -d' ' -f1)
-        version=$(echo "$config_content" | grep -A1 "name: $name" | grep 'version:' | sed -E 's/^\s*version:\s*//')
-        printf "%-31s | %s\n" "$name" "$version"
-    done
-} | tee -a nightly_output.txt
+    # Update monitoring lambda version if specified
+    if [ -n "$MONITORING_LAMBDA_VERSION" ]; then
+        escaped_config_content=$(echo "$escaped_config_content" | yq eval '.MarketplaceItems |= map(select(.name == "unity-cs-monitoring-lambda") |= . * {"version": "'$MONITORING_LAMBDA_VERSION'"})' -)
+    fi
 
-# Escape any special characters in the config content
-escaped_config_content=$(echo "$config_content" | sed 's/"/\\"/g')
+    # Update apigateway version if specified
+    if [ -n "$APIGATEWAY_VERSION" ]; then
+        escaped_config_content=$(echo "$escaped_config_content" | yq eval '.MarketplaceItems |= map(select(.name == "unity-apigateway") |= . * {"version": "'$APIGATEWAY_VERSION'"})' -)
+    fi
+
+    # Update proxy version if specified
+    if [ -n "$PROXY_VERSION" ]; then
+        escaped_config_content=$(echo "$escaped_config_content" | yq eval '.MarketplaceItems |= map(select(.name == "unity-proxy") |= . * {"version": "'$PROXY_VERSION'"})' -)
+    fi
 
+    # Update UI version if specified
+    if [ -n "$UI_VERSION" ]; then
+        escaped_config_content=$(echo "$escaped_config_content" | yq eval '.MarketplaceItems |= map(select(.name == "unity-ui") |= . * {"version": "'$UI_VERSION'"})' -)
+    fi
+    
+    # Log MC version/SHA before deployment
+    echo "Starting CloudFormation deployment with:"
+    echo "  - MC Version: ${MC_VERSION}"
+    echo "  - MC SHA: ${MC_SHA}"
+    echo ""
+
+    # Output the marketplace items table
+    {
+        echo "-----------------------------------------"
+        echo "Items that will auto-deploy on bootstrap:"
+        echo "Marketplace Item                | Version"
+        echo "--------------------------------+--------"
+        yq eval '.MarketplaceItems[] | [.name, .version] | join(" | ")' <(echo "$escaped_config_content") | \
+        while IFS='|' read -r name version; do
+            printf "%-31s |%s\n" "$name" "$version"
+        done
+    } | tee -a nightly_output.txt
+else
+    echo "Config file not found: $CONFIG_FILE"
+    escaped_config_content="[]"
+fi
+
+# Escape any special characters in the config content
+escaped_config_content=$(echo "$escaped_config_content" | sed 's/"/\\"/g')
 
 
 # Modify the CloudFormation create-stack command
 aws cloudformation create-stack \
   --stack-name ${STACK_NAME} \
-  --template-body file://./cloudformation-template/unity-mc.main.template.yaml \
+  --template-body file://../cloudformation-template/unity-mc.main.template.yaml \
```

Comment:
The _cloudformation-template_ directory is now beneath nightly_tests, so this should be a single dot.